src/Microsoft.AspNet.BasicMiddleware/OverrideHeaderMiddlewareOptions.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.BasicMiddleware
+{
+    public class OverrideHeaderMiddlewareOptions
+    {
+        public OverrideHeaderMiddlewareOptions(bool allOverridesEnabled = true)
+        {
+            XForwardedForEnabled = allOverridesEnabled;
+            XForwardedHostEnabled = allOverridesEnabled;
+            XForwardedProtoEnabled = allOverridesEnabled;
+            XHttpMethodOverrideEnabled = allOverridesEnabled;
+        }
+
+        public bool XForwardedForEnabled { get; set; }
```

Comment:
These four boolean properties could be replaced by a [single enum](https://msdn.microsoft.com/en-us/library/system.flagsattribute(v=vs.110).aspx).

This enum type could be used as the ctor param too for greater flexibility.